File: test39_out/atmel.atmega8/Timer0MatchA/Timer0MatchA.js

/*
 * ======== MODULE atmel.atmega8.Timer0MatchA ========
 */

var $$c = function() {
    const Timer0MatchA = {};
    Timer0MatchA.Timer0MatchA = Timer0MatchA
    Timer0MatchA.$name = 'atmel.atmega8.Timer0MatchA';
    Timer0MatchA.pollen$used = 0;
    Timer0MatchA.enable = new $$Ref('atmel_atmega8_Timer0MatchA_enable__E');
    Timer0MatchA.disable = new $$Ref('atmel_atmega8_Timer0MatchA_disable__E');
    Timer0MatchA.clear = new $$Ref('atmel_atmega8_Timer0MatchA_clear__E');
    Timer0MatchA.isEnabled = new $$Ref('atmel_atmega8_Timer0MatchA_isEnabled__E');
    Timer0MatchA.targetInit = new $$Ref('atmel_atmega8_Timer0MatchA_targetInit__I');
    Timer0MatchA.name = "TIMER0_COMPA_vect";
    Timer0MatchA.enableRegister = "TIMSK0";
    Timer0MatchA.enableBit = "OCIE0A";
    Timer0MatchA.clearRegister = "TIFR0";
    Timer0MatchA.clearBit = "OCF0A";
    Timer0MatchA.setHandlerOnHost = function( h ) {
        var $$text = '';
        (Timer0MatchA.handler) = (h);
        return $$text;
    }
    Timer0MatchA.$$hostInit = function() {
        var $$text = '';
        return $$text;
    }
    Timer0MatchA.pollen__uses__ = function() {
    }
    Timer0MatchA.$$privateInit = function() {
    }
    return Timer0MatchA;
}
var $$u = $$c();
$units['atmel.atmega8.Timer0MatchA'] = $$u;
$units.push($$u);

File: test39_out/atmel.atmega8/Timer0MatchA/Timer0MatchA.c

/*
 * ======== string literals ========
 */

static const char atmel_atmega8_Timer0MatchA__s__fileName[] = "Timer0MatchA.p";
static const char atmel_atmega8_Timer0MatchA__s__1[] = "TIMER0_COMPA_vect";
static const char atmel_atmega8_Timer0MatchA__s__3[] = "OCIE0A";
static const char atmel_atmega8_Timer0MatchA__s__5[] = "OCF0A";
static const char atmel_atmega8_Timer0MatchA__s__4[] = "TIFR0";
static const char atmel_atmega8_Timer0MatchA__s__2[] = "TIMSK0";
static const char atmel_atmega8_Timer0MatchA__s__0[] = "Timer0MatchA";

/*
 * ======== code injections ========
 */

 #include <avr/interrupt.h> 

    ISR( TIMER0_COMPA_vect ) {
      (atmel_atmega8_Timer0MatchA_handler__V)();
    }
   

/*
 * ======== function definitions ========
 */

void atmel_atmega8_Timer0MatchA_enable__E() {
    TIMSK0 |= (1 << OCIE0A);
}

void atmel_atmega8_Timer0MatchA_disable__E() {
    TIMSK0 &= ~(1 << OCIE0A);
}

bool atmel_atmega8_Timer0MatchA_isEnabled__E() {
    return(  TIMSK0 & (1 << OCIE0A)  );
}

void atmel_atmega8_Timer0MatchA_clear__E() {
    TIFR0 &= ~(1 << OCF0A);
}

void atmel_atmega8_Timer0MatchA_targetInit__I() {
}

